block/src/header.rs

```diff
@@ -118,83 +97,18 @@ impl Encodable for Header {
         s.append(&self.gas_used);
         s.append(&self.timestamp);
         s.append(&self.extra_data);
-        s.append(&self.mix_hash);
-        s.append(&self.nonce);
     }
-}
 
-impl Decodable for Header {
-    fn decode(rlp: &UntrustedRlp) -> Result<Self, DecoderError> {
-        Ok(Self {
-            parent_hash: rlp.val_at(0)?,
-            ommers_hash: rlp.val_at(1)?,
-            beneficiary: rlp.val_at(2)?,
-            state_root: rlp.val_at(3)?,
-            transactions_root: rlp.val_at(4)?,
-            receipts_root: rlp.val_at(5)?,
-            logs_bloom: rlp.val_at(6)?,
-            difficulty: rlp.val_at(7)?,
-            number: rlp.val_at(8)?,
-            gas_limit: rlp.val_at(9)?,
-            gas_used: rlp.val_at(10)?,
-            timestamp: rlp.val_at(11)?,
-            extra_data: rlp.val_at(12)?,
-            mix_hash: rlp.val_at(13)?,
-            nonce: rlp.val_at(14)?,
-        })
+    pub fn partial_hash(&self) -> H256 {
```

Comment:
This looks a lot like `UnsignedTransaction::signing_hash`... because it must implemented for both interfaces? And I guess it's `partial` because the Header can be signed or not?

Reply:
This line is to get the partial header hash used in ethash's hashimoto, which will then validate nonce and mix hash. Yep it looks a lot like signing_hash, but probably not quite possible to remove this duplicate because transaction's signing hash uses a different "partial" logic.

This was once in a struct called `PartialHeader`. Decided to remove it because the only place that this partial header is used is to derive the hash. The same is for signing_hash.

Reply:
⚡️ 👍